### PR TITLE
feat(cb2-9056): filter first test / first test retest 

### PR DESCRIPTION
**Pull request by JunYanBJSS**
## Make it harder to incorrectly select a first test / first test retest
add a filter to filterTestType to meet the AC. Fixed all of failed test .
[CB2-9056](https://dvsa.atlassian.net/browse/CB2-9056)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/test-records/amend/views/test-type-select-wrapper/test-type-select-wrapper.component.spec.ts [hunk 1]
```diff
@@ -16,7 +19,7 @@ describe('TestTypeSelectWrapperComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [TestTypeSelectWrapperComponent, TestTypeSelectComponent],
-      imports: [RouterTestingModule],
+      imports: [RouterTestingModule, HttpClientTestingModule, StoreModule.forRoot({ technicalRecord: vehicleTechRecordReducer })],
       providers: [{ provide: TestTypesService, useValue: { selectAllTestTypes$: of([]), testTypeIdChanged: () => {} } }],
```
Comment (pbardy2000) [inline src/app/features/test-records/amend/views/test-type-select-wrapper/test-type-select-wrapper.component.spec.ts:23]:
```suggestion
      imports: [RouterTestingModule, HttpClientTestingModule],
      providers: [
          provideMockStore<State>({ initialState: initialAppState }),
          { provide: TestTypesService, useValue: { selectAllTestTypes$: of([]), testTypeIdChanged: () => {} } }
      ],
```
Generally better practice to mock dependencies for unit tests as you only want to test your specific changes in isolation - errors in these may make it seem like your code is broken, when it isn't.

Comment (JunYanBJSS) [reply src/app/features/test-records/amend/views/test-type-select-wrapper/test-type-select-wrapper.component.spec.ts]: Yeah, you are right. I am not pretty much sure about the unit test. Thank you so much for pointing it out to me. I have amended it now.

**Review by pbardy2000 — APPROVED** [review]

File: src/app/features/test-records/components/test-type-select/test-type-select.component.ts [hunk 2]
```diff
@@ -8,12 +12,23 @@ import { Observable } from 'rxjs';
...
-  constructor(private testTypesService: TestTypesService) {}
+  constructor(
+    private testTypesService: TestTypesService,
+    private technicalRecordService: TechnicalRecordService,
+    private router: Router,
+  ) {}
+  ngOnInit(): void {
+    this.technicalRecordService.techRecordHistory$.subscribe((recordHistory) => {
+      if (!recordHistory) {
+        void this.router.navigate(['tech-records/:systemNumber/:createdTimestamp']);
```
Comment (shivangidas) [inline src/app/features/test-records/components/test-type-select/test-type-select.component.ts:28]: The systemNumber or createdTimestamp needs to be added

Comment (JunYanBJSS) [reply src/app/features/test-records/components/test-type-select/test-type-select.component.ts:28]: Nice shout-out. I will add it now.

File: src/app/store/test-types/selectors/test-types.selectors.ts [hunk 3]
```diff
@@ -25,12 +27,20 @@ export const selectTestTypesTotal = createSelector(testTypesFeatureState, (state
 
 export const selectTestTypesLoadingState = createSelector(testTypesFeatureState, (state) => state.loading);
 
-export const selectTestTypesByVehicleType = createSelector(selectAllTestTypes, toEditOrNotToEdit, (testTypes, testResult) => {
-  if (testResult) {
-    return filterTestTypes(testTypes, testResult);
-  }
-  return [];
-});
+// eslint-disable-next-line max-len
```
Comment (shivangidas) [inline src/app/store/test-types/selectors/test-types.selectors.ts:30]: don't think this disable is required
```suggestion

```

File: src/app/store/test-types/selectors/test-types.selectors.ts [hunk 4]
```diff
@@ -25,12 +27,20 @@ export const selectTestTypesTotal = createSelector(testTypesFeatureState, (state
...
+export const selectTestTypesByVehicleType = createSelector(
+  selectAllTestTypes,
+  toEditOrNotToEdit,
+  selectTechRecordHistory,
+  (testTypes, testResult, selectRecordHistory) => {
```
Comment (shivangidas) [inline src/app/store/test-types/selectors/test-types.selectors.ts:35]: can we change the name from selectRecordHistory to techRecordHistory

Comment (JunYanBJSS) [reply src/app/store/test-types/selectors/test-types.selectors.ts:35]: definitely, I have changed it now

File: src/app/store/test-types/selectors/test-types.selectors.ts (hunk 3 above)
Comment (JunYanBJSS) [reply src/app/store/test-types/selectors/test-types.selectors.ts:30]: removed now.

**Review by pbardy2000 — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**